src/all/novelfull/src/eu/kanade/tachiyomi/extension/all/novelfull/NovelFullGlobal.kt

```diff
@@ -0,0 +1,241 @@
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import org.jsoup.select.Evaluator
+import rx.Observable
+
+abstract class NovelFullGlobal(
+    override val lang: String,
+    val supportsSearch: Boolean = true,
+) : ParsedHttpSource() {
+
+    override val baseUrl = "https://allwebnovel.com" // https://mangafre.com
+
+    override val name = "NovelFull"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder().build()
+
+    // ============================== Popular ===============================
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/comic/bookclass.html?type=hot_novel&page_num=$page&language=$lang")
+    }
+
+    override fun popularMangaSelector(): String {
+        return "#list-page > .col-truyen-main .row"
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            title = element.select(Evaluator.Tag("a")).attr("alt")
+            thumbnail_url = element.selectFirst(Evaluator.Tag("img"))!!.imgAttr()
+            setUrlWithoutDomain(element.select(Evaluator.Tag("a")).attr("href"))
+        }
+    }
+
+    override fun popularMangaNextPageSelector(): String? {
+        return ".pagination > .next"
+    }
+
+    // =============================== Latest ===============================
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/comic/bookclass.html?type=last_release&page_num=$page&language=$lang")
+    }
+
+    override fun latestUpdatesSelector(): String = popularMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector(): String? = popularMangaNextPageSelector()
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            return GET("$baseUrl/comic/search.html?keyword=$query&page_num=$page")
+        }
+        filters.forEach { filter ->
+            when (filter) {
+                is GenreFilter -> {
+                    val filterId = filter.selected
+                    return GET("$baseUrl/comic/bookclass.html?type=category_novel&id=$filterId&page_num=$page&language=$lang")
+                }
+                else -> { }
+            }
+        }
+        TODO("Need to implement empty search somewhere")
```

Comment:
And here too

Reply:
just return simple `/comic/bookclass.html`?